### DISCUSSION TITLE
how to create a table via supabase api in  DART ?

### DISCUSSION THREAD

**Discussion by ClarissaGazalaEvanthe** [supabase/supabase#19866] [Questions]
I want to change and create a table using API, currently I am still using manual / writing scripts in the SQL Supabase dashboard tab

**Comment by GaryAustin1** [comment]
The REST API's cannot execute SQL.  Your only choice with them is to use an rpc called Postgres function.  Otherwise you have to use a Postgres library on serverside code to access the direct database port.

**Reply by ClarissaGazalaEvanthe** [reply]
I tried the rpc function, but it didn't work

is there an example?

**Reply by GaryAustin1** [reply]
Did you write a Postgres function to go with the rpc call?   All the code below does is call a Postgres function.  You have to create that function.

There is no info on how to do this in the Supabase docs as it is all a Postgres thing and is frankly not considered good practice with a relational database to create tables on demand unless you have very specific needs.

You will need to search the internet for creating a table with a postgres function or something similar.
There are possibly a few discussions here also.
I found this one that shows code to create a table in a function, although it is in a trigger function, but that is similar.
https://github.com/orgs/supabase/discussions/3852#discussioncomment-5834062

**Comment by ClarissaGazalaEvanthe** [comment]
I tried this but couldn't
```dart
import 'package:supabase/supabase.dart';

void main(List<String> args) async {
  SupabaseClient supabase = SupabaseClient(
    "url",
    "supabase-key",
  );
  final response = await supabase.rpc(
    'create_table',
    params: {
      'table_name': 'your_table_name',
    },
  );
  print(response);
}
```

but an error appears

```bash
Unhandled exception:
PostgrestException(message: Could not find the function public.create_table(table_name) in the schema cache, code: PGRST202, details: Searched for the function public.create_table with parameter table_name or with a single unnamed json/jsonb parameter, but no matches were found in the schema cache., hint: null)
#0      PostgrestBuilder._parseResponse (package:postgrest/src/postgrest_builder.dart:347:7)
#1      PostgrestBuilder._execute (package:postgrest/src/postgrest_builder.dart:202:14)
<asynchronous suspension>
#2      PostgrestBuilder.then (package:postgrest/src/postgrest_builder.dart:424:24)
<asynchronous suspension>
```

**Comment by ClarissaGazalaEvanthe** [comment]
I got the code from AI Supabase

**Comment by ClarissaGazalaEvanthe** [comment]
Please anyone give me example code for changing, adding, deleting, and editing tables using API
